dali/pipeline/operators/reader/loader/file_loader.cc

```diff
@@ -88,7 +88,9 @@ vector<std::pair<string, int>> filesystem::traverse_directories(const std::strin
   std::sort(entry_name_list.begin(), entry_name_list.end());
   for (unsigned dir_count = 0; dir_count < entry_name_list.size(); ++dir_count) {
       assemble_file_list(file_root, entry_name_list[dir_count], dir_count, &file_label_pairs);
-  }
+  } 
+  // sort file names as well
+  std::sort(file_label_pairs.begin(), file_label_pairs.end());
```

Comment:
Why sorting them here and by filename? 
And if we agree about sorting, wouldn’t sorting them by label be more useful?

Reply:
I wanted to be consistent with PyTorch loader. We need to sort by folder name to be sure that 'a' dir gets label 0, and by file name to be sure that dirst sample with label 0 is 'a/a'.
It will make easier comparing data between Dali and no DALI

Reply:
Ok I see